### ISSUE TITLE
Reformatting code is left-aligning comments.

### ISSUE THREAD

**Issue by EBatTiVo** [TiVo/intellij-haxe#295]
Highlight and reformat (Code->Reformat Code) any code containing a comment will place the single-line comments against the left-most column (removes indenting) or will left-align the first line of a block comment.  Copy and paste will left align comments in the pasted block as well.

TIR: STB-10533

**Comment by wiggin77** [comment]
This also happens for me when I type "/**" then hit enter to auto generate a comment block.  First line is always at column 0.  (IntelliJ 14.1.3, plugin 0.9.4)  Btw, same operation works fine for Java.

**Comment by EBatTiVo** [comment]
The code that does reformatting is IDEA's DefaultCodeFormatter (IIRC).  The PSI for comments is likely not what the formatter is expecting.  I suspect, but I haven't yet confirmed, that the change for #209 is causing the issue.

Incidentally, the unit tests do contain this scenario and are NOT failing.  Therefore, I'm half-expecting some secondary processing step to be re-aligning incorrectly.  Stepping (debugging) through unit test code did not reveal anything unusual.

**Comment by as3boyan** [comment]
https://github.com/TiVo/intellij-haxe/blob/ImplementNestedFunctionTypesSupport/src/common/com/intellij/plugins/haxe/ide/formatter/HaxeIndentProcessor.java#L59-59

**Comment by as3boyan** [comment]
If I got it right, then I think that it works in a proper way, single line comments in IDEA added from line start, so no problem with that, but if it's added in some other IDE, then this may happen.

**Comment by as3boyan** [comment]
@wiggin77 /** indeed works wierd, but /* works just fine

**Comment by EBatTiVo** [comment]
> If I got it right, then I think that it works in a proper way, single line comments in IDEA added from line start, so no problem with that, but if it's added in some other IDE, then this may happen.

I don't understand.  Are you saying that single line comments *should* be aligned on the left-most column (column 1) when the code is reformatted?  If so, then I disagree.

Right now, this:
```haxe
class Test {
    // a comment
    public function new() {}
}
```
is reformatted to this:
```haxe
class Test {
// a comment
    public function new() {}
}
```

If the comment started in the left-most column (column 1), then it should stay there.  If the comment has *any* indentation, then it should be indented to match the code.

**Comment by soywiz** [comment]
+1 I had that problem. All my comments were moved to the left-most column and it was a bit annoying.

**Comment by as3boyan** [comment]
I checked how it works in Java indeed it works differently

Java:
![Imgur](http://i.imgur.com/sSzjjC2.png)

Notice where it inserts comment (// part)
Always indents code on enter

Haxe:
![Imgur](http://i.imgur.com/nD6UVhc.png)
If I hit enter after method body - indentation is wrong here
Also on comment (//) gets inserted at the beginning of line

**Comment by EBatTiVo** [comment]
Fix is in #372.  Awaiting review.

**Comment by EBatTiVo** [comment]
Fixed with #372 and merged.
